vyper/builtin_functions/functions.py

```diff
@@ -851,8 +855,12 @@ def build_LLL(self, expr, args, kwargs, context):
         elif sub.location == "storage":
             lengetter = LLLnode.from_list(["sload", "_sub"], typ=BaseType("int128"))
             elementgetter = _storage_element_getter
-        # TODO: unclosed if/elif clause.  Undefined behavior if `sub.location`
-        # isn't one of `memory`/`storage`
+        elif sub.location == "calldata":
+            lengetter = LLLnode.from_list(["calldatasize"], typ="uint256")
```

Comment:
Sorry I know I looked at this before but I must have missed this. I don't think this is actually correct, since downstream code assumes that the bytestring has a 32-byte header.

Here's the code generated for `extract32(msg.data, 0)`:
```
                    [calldataload, [mul, 32, [div, [clample, 0, [sub, calldatasize, 32]], 32]]]]],
```

I think the kludge would be to return `(add 32 calldatasize)` for `lengetter`. The actual fix would be to refactor Extract32 so it's less broken. I can chat with you about my ideas for that, if you are interested.

Reply:
it took me a moment but it clicked, I've implemented the quick fix for the length getter `(add 32 calldatasize)`